ibverbs/src/lib.rs

```diff
@@ -769,14 +804,57 @@ pub struct PreparedQueuePair<'res> {
 /// An identifier for the network endpoint of a `QueuePair`.
 ///
 /// Internally, this contains the `QueuePair`'s `qp_num`, as well as the context's `lid` and `gid`.
-#[derive(Copy, Clone)]
+#[derive(Clone)]
 pub struct QueuePairEndpoint {
     num: u32,
     lid: u16,
     gid: ffi::ibv_gid,
 }
 
-impl<'res> PreparedQueuePair<'res> {
+use serde::{Deserialize, Serialize};
+
+/// A wrapper around `QueuePairEndpoint` with addintional `rkey` and `raddr` field. Intended to be sent to the peer.
+// TODO: write more thoughout documentation for fields.
+#[derive(Serialize, Deserialize, Clone)]
+pub struct EndpointMsg {
```

Comment:
We can probably make this nicer with the changes that landed in #7.

Reply:
Oh! That's some good timing